app/src/test/java/com/x8bit/bitwarden/data/platform/repository/ServerConfigRepositoryTest.kt

```diff
@@ -0,0 +1,190 @@
+package com.x8bit.bitwarden.data.platform.repository
+
+import app.cash.turbine.test
+import com.x8bit.bitwarden.data.platform.base.FakeDispatcherManager
+import com.x8bit.bitwarden.data.platform.datasource.disk.model.ServerConfig
+import com.x8bit.bitwarden.data.platform.datasource.disk.util.FakeConfigDiskSource
+import com.x8bit.bitwarden.data.platform.datasource.network.model.ConfigResponseJson
+import com.x8bit.bitwarden.data.platform.datasource.network.model.ConfigResponseJson.EnvironmentJson
+import com.x8bit.bitwarden.data.platform.datasource.network.model.ConfigResponseJson.ServerJson
+import com.x8bit.bitwarden.data.platform.datasource.network.service.ConfigService
+import com.x8bit.bitwarden.data.platform.manager.dispatcher.DispatcherManager
+import com.x8bit.bitwarden.data.platform.repository.model.Environment
+import com.x8bit.bitwarden.data.platform.repository.util.FakeEnvironmentRepository
+import com.x8bit.bitwarden.data.platform.util.asSuccess
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.test.runTest
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertNotEquals
+import org.junit.jupiter.api.Assertions.assertNotNull
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import java.time.Instant
+
+class ServerConfigRepositoryTest {
+    private val fakeDispatcherManager: DispatcherManager = FakeDispatcherManager()
+    private val fakeConfigDiskSource = FakeConfigDiskSource()
+    private val configService: ConfigService = mockk {
+        coEvery {
+            getConfig()
+        } returns CONFIG_RESPONSE_JSON.asSuccess()
+    }
+    private val environmentRepository = FakeEnvironmentRepository().apply {
+        environment = Environment.Us
+    }
+
+    @BeforeEach
+    fun setUp() {
+        fakeConfigDiskSource.serverConfig = null
+    }
+
+    private val repository = ServerConfigRepositoryImpl(
+        configDiskSource = fakeConfigDiskSource,
+        configService = configService,
+        environmentRepository = environmentRepository,
+        dispatcherManager = fakeDispatcherManager,
+    )
+
+    @Test
+    fun `environmentRepository stateflow should trigger new server configuration`() = runTest {
+        val serverConfigBefore = fakeConfigDiskSource.serverConfig
+
+        // This should trigger a new server config to be fetched
+        environmentRepository.environment = Environment.Eu
+
+        repository.serverConfigStateFlow.test {
+            val newConfig = awaitItem()
+            assertNotEquals(
+                serverConfigBefore,
+                newConfig,
+            )
+
+            assertEquals(
+                SERVER_CONFIG.serverData,
+                newConfig!!.serverData,
```

Comment:
Can we just validate the entire `newConfig`, instead of comparing a partial update.

Reply:
With the fixed clock I believe now we can, let me change that.